Update config instructions (#56)
LycheeOrg/LycheeOrg.github.io: docs/configuration.md

```
> {tip} You may customize the default maintenance mode template by defining your own template at `resources/views/errors/503.blade.php`.

## Configuration options

|Name|Description|Default|
|---|---|---|
|`APP_NAME`|The gallery name|`Lychee`|
|`APP_ENV`|Environment of your gallery, `production` or `development`|`production`|
|`APP_DEBUG`|Enable debug mode|`false`|
|`APP_URL`|The URL of your gallery|`http://localhost`|

> |`APP_KEY`|Your app key, a 32 char string (set during installation)|`null`|

Reviewer: ```suggestion
|`APP_KEY`|Your app key, a 32 char string (set during installation) which is used for encryption|`null`|
```
?

Reviewer 2: ```suggestion
|`APP_KEY`|Your app key which is used for encryption (set during installation)|`null`|
```

I'm fine with the addition, but I would leave out the 32 char string. On my installation it is a base 64-encoded random value with 48 characters. I assume the information about 32 characters is outdated.

Reviewer: In that case.... *goes elsewhere...*

> |`TIMEZONE`|The timezone of your server|system timezone of server|

Reviewer 3: Shouldn't the description of this one be more like "The timezone of your photos"?

|`DEBUGBAR_ENABLED`|Enable debugbar (only for debugging)|`false`|
|`LOG_CHANNEL`|Default log channel, `single`, `daily`, `slack`, `syslog`, `errorlog`, `monolog`, `custom` or `stack`|`stack`|
|`DB_CONNECTION`|Database type, `mysql`, `pgsql`, `sqlite` or `sqlsrv` (`sqlsrv` is not supported officially and may not work)|`mysql`|
|`DB_DATABASE`|Path to SQLite database or name of MySQL/PostgreSQL DB|`database/database.sqlite` or `forge`|
|`DB_HOST`|Host of DB server|`127.0.0.1`|
|`DB_PORT`|Post of DB server|`3306` (MySQL) or `5432` (PostgreSQL)|
|`DB_USERNAME`|Username of database|`forge`|
|`DB_PASSWORD`|Password of database user|empty string|

> |`DB_OLD_LYCHEE_PREFIX`|Table prefix of old Lychee database|empty string|

Reviewer 3: Should we say something more here rather than just "old"? Maybe add "(v3)"?

|`DB_LOG_SQL`|Log SQL statements (only use for debugging)|`false`|

> |`BROADCAST_DRIVER`|The broadcaster used if the framework needs to broadcast an event, `pusher`, `redis`, `log` or `null`|`null`|

Reviewer 3: Do we use this for anything? If not, then I suppose we could remove this one as well as `PUSHER_*` and `REDIS_*` variables below?

Reviewer 2: The question was not whether Redis can be used by some other services, but if we use the broadcasting services. We don't.

|`PUSHER_APP_KEY`|App key for the `pusher`|`null`|
|`PUSHER_APP_SECRET`|App secret for the `pusher`|`null`|
|`PUSHER_APP_ID`|App ID for the `pusher`|`null`|
|`PUSHER_APP_CLUSTER`|App cluster for the `pusher`|`null`|

> |`CACHE_DRIVER`|The driver used for caching, `apc`, `array`, `database`, `file`, `memcached`, `redis` or `dynamodb`|`file`|

Reviewer 3: Same question: does this do anything in Lychee?

Reviewer 2: The cache is the session cache. Yes, of course it is used. This is where the session ID and the authenticated user ID are stored between two consecutive requests.

|`REDIS_HOST`|Redis host|`127.0.0.1`|
|`REDIS_PASSWORD`|Redis password|`null`|
|`REDIS_PORT`|Redis port|`6379`|

> |`LYCHEE_UPLOADS`|Path to uploads directory|`uploads/`|

Reviewer 3: Isn't the default `public/uploads/`?

Reviewer 2: No. The prefix `public` is added later by something like `public_path(...)`. The common prefix `public` is configured somewhere else independently.

|`LYCHEE_UPLOADS_URL`|URL to uploads directory|`uploads/`|

> |`LYCHEE_DIST`|Path to dist directory|`dist/`|

Reviewer 3: Again, shouldn't this be `public/dist`?

Reviewer 2: See the previous answer.

|`LYCHEE_DIST_URL`|URL to dist directory|`dist/`|
|`MAIL_DRIVER`|Mailer type, `smtp`, `ses`, `mailgun`, `postmark`, `sendmail`, `log` or `array`|`smtp`|
|`MAIL_HOST`|Host of SMTP server|`smtp.mailgun.org`|
|`MAIL_PORT`|Port of SMTP server|`587`|
|`MAIL_ENCRYPTION`|Encryption for SMTP server|`tls`|
|`MAIL_USERNAME`|Username of SMTP server|`null`|
|`MAIL_PASSWORD`|Password of SMTP server|`null`|
|`MAIL_FROM_ADDRESS`|"From" address|`hello@example.com`|

> |`MAIL_FROM_ADDRESS`|"From" name|`Example`|

Reviewer 3: ```suggestion
|`MAIL_FROM_NAME`|"From" name|`Example`|
```

> |`QUEUE_CONNECTION`|Connection for queues, `sync`, `database`, `beanstalkd`, `sqs` or `redis`|`sync`|

Reviewer 3: Does that do anything for us?

Reviewer 2: No it does not. We don't use the queuing service at all. In theory, we could completely nuke it from the `AppServiceProvider` and all of its configuration. The idea of the queuing service is to create asynchronous tasks, which can be put into a queue of pending tasks and then the system's cron job picks of the task to run.

Reviewer: Something makes me think that the email notifications use Laravel's cron. They may well use some other path to get there though.

Reviewer 2: The mail notification don't use the queuing system. They could, but the don't. Instead the mailing system is a simply CLI command. This CLI command is directly invoked by the system's cron job without the indirection provided by the queuing system.

|`SESSION_DRIVER`|Driver for sessions, `file`, `cookie`, `database`, `apc`, `memcached`, `redis`, `dynamodb` or `array`|`file`|

> |`SESSION_LIFETIME`|Session lifetime in minutes|`120`|

Reviewer 3: Maybe add a bit more of an explanation about this one? Say, "Idle session timeout in minutes; the session will need to be reinitialized once it expires"?

|`TRUSTED_PROXIES`|Trusted proxy IP addresses|`null`|
|`SECURITY_HEADER_HSTS_ENABLE`|Enable HTTP strict transport security|`false`|

> |`SESSION_SECURE_COOKIE`|Cookies only via HTTPS|`null`|

Reviewer 3: ```suggestion
|`SESSION_SECURE_COOKIE`|Cookies only via HTTPS|`false`|
```

|`LOG_SLACK_WEBHOOK_URL`|Webhook URL for slack logging|`null`|
|`PAPERTRAIL_URL`|URL for papertrail logging|`null`|
|`PAPERTRAIL_PORT`|Port for papertrail logging|`null`|
|`LOG_STDERR_FORMATTER`|Formatter for logging to stderr|`null`|
|`MAIL_LOG_CHANNEL`|Log channel for mails|`null`|
|`REDIS_QUEUE`|Redis queue|`default`|
|`SQS_PREFIX`|SQS prefix|`https://sqs.us-east-1.amazonaws.com/your-account-id`|
|`SQS_QUEUE`|SQS queue name|`your-queue-name`|
|`SQS_SUFFIX`|SQS suffix|`null`|
|`QUEUE_FAILED_DRIVER`|driver for failed queues|`database-uuids`|
|`MAILGUN_DOMAIN`|Domain to mailgun|`null`|
|`MAILGUN_SECRET`|Serect for mailgun|`null`|
|`MAILGUN_ENDPOINT`|Endpoint to mailgun|`api.mailgun.net`|
|`POSTMARK_TOKEN`|Token for postmark|`null`|
|`SESSION_CONNECTION`|Session connection for `database` and `redis` drivers|`null`|
|`SESSION_STORE`|Store for sessions, affects `apc`, `dynamodb`, `memcached` and `redis` drivers|`null`|
|`SESSION_COOKIE`|The cookie's name|the value of `APP_NAME` + `_session`|
|`SESSION_DOMAIN`|Session cookie domain|`null`|
|`SESSION_SECURE_COOKIE`|Cookies only via HTTPS|`null`|
|`VIEW_COMPILED_PATH`|Where to store compiled Blade templates|`framework/views`|

> 

Reviewer: I wonder whether we should change our defaults to match the example. In theory we could move all the documentation of it there, but we change it rarely enough that being in two places shouldn't be a major issue. We'll have to remember this exists though.

Reviewer 2: I wonder if we should really list all the configuration options explicitly. While I appreciate the effort it will soon get outdated.

Reviewer 3: Do all of these actually _do_ anything for Lychee? Say, what is `ASSET_URL` used for? I'm of the opinion that if a setting simply doesn't apply to Lychee at all, we should skip it.

|`APP_NAME`|The gallery name|`Lychee`|
|`APP_ENV`|Environment of your gallery, `production` or `development`|`production`|
|`APP_DEBUG`|Enable debug mode|`false`|

> |`APP_URL`|The URL of your gallery (under which the `public/` folder is accessible)|http://localhost`|

Reviewer 2: One of the native speakers (@kamil4 or @d7415) should have a look at this one. The preposition "under" sounds very German to my ears. I  would have written something like

```suggestion
|`APP_URL`|The URL of your gallery (which resolves to the `public/` folder)|http://localhost`|
```

But there are probably better alternatives.

Reviewer: I prefer yours to the original. Nothing better is coming to mind.

|`MAIL_USERNAME`|Username of SMTP server|`null`|
|`MAIL_PASSWORD`|Password of SMTP server|`null`|
|`MAIL_FROM_ADDRESS`|"From" address|`hello@example.com`|
|`MAIL_FROM_NAME`|"From" name|`Example`|
|`SESSION_DRIVER`|Driver for sessions, `file`, `cookie`, `database`, `apc`, `memcached`, `redis`, `dynamodb` or `array`|`file`|

> |`SESSION_LIFETIME`|Idle session timeout in minutes; the session will need to be reinitialized once it expires|`120`|

Reviewer 2: ```suggestion
|`SESSION_LIFETIME`|Idle session expiration in minutes; the session will need to be reinitialized once it has expired|`120`|
```

Please don't use the word "timeout". I know that "timeout" and "expiration" can mostly be used as synonyms, but this one has already led me on a wrong path once. In a technical context, for me a request/response pair is something which can time out, but a session can expire.

Moreover, I am not sure if the "will" is 100% correct. According to my schoolbook English it is not, because this is nothing which will happen in an indefinite future, but more a mathematical condition which always holds so it should use present tense. But native speakers to the front (@kamil4 and @d7415).

Reviewer: "will" is fine. "the session has to be reinitialised when it expires" would also work, but I wouldn't say it was better.
